db/migrate/20210414203512_convert_primary_language_to_iso_code.rb

```diff
@@ -0,0 +1,147 @@
+    "bosnian": ['bos'],
+    "Chinese": ['Chinese (Cantonese)', 'Chinese (Mandarin)', 'Chinese (not specified)', 'Mandarin', 'manderin chinese'],
+    "English": ['E', 'ENGLIAH', 'ENHLISH', 'Emglish', 'Engiish', 'Engish', 'Englaih', 'England', 'Englislh', 'Englsh', 'Englsih', 'Englxish', 'Enlish',
+                'dad limited eng', 'parent eng'],
+    "French": ['FR', 'fre', 'lingala, french'],
+    "filipino": ['philippines'],
+    "frisian": ['frysian'],
+    "haitian": ['haitian creole', 'HAITIAN FRENCH CREOLE', 'Haitian Creole'],
+    "Lao": %w[Laos Laotian],
+    "Marshallese": ['marshalee'],
+    "Panjabi": ['punjabi'],
+    "Persian": ['Persian (Farsi)', 'farsi'],
+    "Portuguese": ['PORTG', 'PORTUG', 'lingala/portuguese', 'portguese', 'portugese'],
+    "Somali": ['somali, may maay', 'somali, may-maay'],
+    "Spanish": ['Espanol', 'SPAN', 'SpanishA', 'dad span'],
+    "Telugu": %w[pelugu Telegu],
+    "Vietnamese": ['vietamese'],
+    'nil': [' French-based (Other)', '1', '9', 'Chin', 'Creoles and pidgins', 'Dari', 'Jefferson', 'NA', 'No.', 'None', 'Other', 'RUNYORO', 'UND',
+            'Undetermined', 'Unknown', 'afro-asiatic languages', 'no']
+  }.freeze
+
+  # For some translations we want to leave comments explaining the translation
+  # For example, it doesnt make sense to tell the User that we've translated "ENGLLISH" to "English"
+  # But if we translate 'dad span/eng' to spanish and english, it's worth telling them in a note
+  # Only for values listed below will the system create History items explaining the language mapping
+  TRANSLATION_COMMENTS = [' French-based (Other)', 'American Sign Language', 'Chin', 'Chinese (Cantonese)', 'Chinese (Mandarin)',
+                          'Chinese (not specified)', 'Creoles and pidgins', 'Dari', 'HAITIAN FRENCH CREOLE', 'Haitian Creole', 'Jefferson', 'Laos',
+                          'Laotian ', 'Mandarin', 'RUNYORO', 'Sign Languages', 'acholi', 'afro-asiatic languages', 'dad limited eng', 'dad span',
+                          'dad span/eng', 'eng/braz', 'farsi', 'lingala, french', 'lingala/portuguese', 'manderin chinese', 'no', 'parent eng',
+                          'somali, may maay', 'somali, may-maay'].freeze
+  def up
+    add_column :patients, :legacy_primary_language, :string
+    add_column :patients, :legacy_secondary_language, :string
+    execute <<-SQL.squish
+        UPDATE patients
+        SET legacy_primary_language = primary_language, legacy_secondary_language = secondary_language
+        WHERE purged = FALSE
+    SQL
+
+    # automatic_translations will take the format of { 'English' : 'eng' }
+    # Where `English` is plucked directly from the DB
+    automatic_translations = {}
+
+    # custom_translations will take the format of { 'engish' : 'English', 'englsh' : 'English'...etc }
+    # It is merely a re-formatting of the TRANSLATION_LIST above
+    custom_translations = {}
+    TRANSLATION_LIST.each do |key, value|
+      key = key == :nil ? nil : key.to_s
```

Comment:
looks like `key` is already a string, so maybe we can just say `key = nil if key == :nil`

Reply:
It isnt, so i'll keep this as is!